src/main.rs

```diff
@@ -48,47 +50,59 @@ fn main() -> Result<(), Box<dyn Error>> {
         }
     } else if let Some(input) = cli.input {
         let input = String::from(input.to_string_lossy());
+        let code = {
+            if input == "-" || input == "--silence" {
+                let mut buf = String::new();
+                match stdin().read_to_string(&mut buf) {
+                    Ok(_) => buf,
+                    Err(err) => handle_err(err)
+                }
+            } else {
+                match fs::read_to_string(&input) {
+                    Ok(code) => code,
+                    Err(err) => handle_err(err)
+                }
+            }
+        };
+        match AmberCompiler::new(code, Some(input)).compile() {
+            Ok((messages, code)) => {
+                messages.iter().for_each(|m| m.show());
+                // Save to the output file
+                if let Some(output) = cli.output {
+                    let outs = String::from(output.to_string_lossy());
+                    if outs == "/dev/null" {
```

Comment:
This should have been the `--silent`

```suggestion
                    if outs == "--silent" {
```